tests/test_io.py

```diff
@@ -1690,3 +1691,378 @@ def test_csv_options_cs2(self):
+
+        fmu.setup_experiment()
+        fmu.initialize()
+        opts["initialize"] = False
+
+        diag_params, diag_vars = setup_diagnostics_variables(
+            model = fmu,
+            start_time = 0,
+            options = opts,
+            solver_options = opts['CVode_options'])
+
+        result_handler.set_options(opts) # required in order to call simulation_start()
+        result_handler.initialize_complete()
+        if opts["dynamic_diagnostics"]:
+            result_handler.simulation_start(diag_params, diag_vars)
+        else:
+            result_handler.simulation_start()
+        if not only_diagnostics_data:
+            result_handler.integration_point()
+
+        fmu.time += 0.01
+        fmu.set('J4.phi', math.sin(5*fmu.time + math.pi/2)) # arbitrary
+
+        rdb = ResultDymolaBinary(opts["result_file_name"], allow_file_updates=True)
+
+
+        if only_diagnostics_data:
+            vars_to_test = ['@Diagnostics.step_time', '@Diagnostics.nbr_steps']
+        else:
+            vars_to_test = ['time', 'J4.phi']
+            if opts['dynamic_diagnostics']:
+                vars_to_test += ['@Diagnostics.step_time', '@Diagnostics.nbr_steps']
+
+        nbr_of_calls = 10
+        counter = 0
+        diag_data_ratio = np.random.randint(2, 5) # random for testing purpose
+        start_index = 0
+
+        for i in range(nbr_of_calls):
+
+            nbr_of_points = np.random.randint(1, 20)
+            h = 1/nbr_of_points * 0.1 # "arbitrary" step through time
+
+            for j in range(nbr_of_points):
+
+                if favor_diagnostics_data:
+                    if counter%diag_data_ratio == 0:
+                        if not only_diagnostics_data:
+                            result_handler.integration_point()
+                        result_handler.integration_point()
+                    if opts["dynamic_diagnostics"]:
+                        diag_vars['@Diagnostics.step_time'] = (h, 'Step time') # arbitrary test value
+                        result_handler.diagnostics_point(np.array([val[0] for val in diag_vars.values()], dtype=float))
+                else:
+                    if opts["dynamic_diagnostics"] and counter%diag_data_ratio==0:
+                        diag_vars['@Diagnostics.step_time'] = (h, 'Step time') # arbitrary test value
+                        result_handler.diagnostics_point(np.array([val[0] for val in diag_vars.values()], dtype=float))
+                    if not only_diagnostics_data:
+                        result_handler.integration_point()
+
+                counter += 1
+                fmu.set('J4.phi', math.sin(5*fmu.time + math.pi/2)) # arbitrary
+                fmu.time += h
+            trajectories, new_start_index = rdb.get_variables_data(vars_to_test, start_index, None)
+        if update_start_index:
+            start_index = new_start_index
+
+    @testattr(stddist = True)
+    def test_get_variables_data_0(self):
```

Comment:
I might be missing something here, but what exactly is being tested here?

The helper function just generated data, there are no asserts?

Also, I think it would be useful if the naming/docstrings would describe the practical scenario being tested, e.g., low/high frequency diagnostics, continuous fetching, etc...

Reply:
The first sets of test just verifies that we can retrieve the data, when the test scenario is constructed in different ways, i.e.:
1. No diagnostics data
2. With diagnostics data
3. More diagnostics data than regular data
4. Only diagnostics data but no regular data
There was also some randomness added here because we wanted to make sure it wasn't something as "exactly half the amount of diagnostics data as regular data".
We don't check the values in these tests since we tests that check values further down. I see why it might look a bit redundant though so let me know if you think we should just remove those tests.
When developing the tests however it was helping to make sure we don't have segmentation faults for rare cases where the ratio between regular data and diagnostics data would for some reason end up crashing together with different inputs for `start_index` and `stop_index`.

Reply:
1. I don't fancy randomness here since it adds the risk that for (un)fortunate values a test may erroneously pass when it should not. Sure, we cannot test all combinations of integration vs. diagnostics points, but I'd still prefer having fixed and/or deterministic ratios, so we know some important cases are covered. There will always be a limitation on how many you can practically test.
2. To me it sounds like the first set of tests that only generate the data is just a single step in the second test and you kept it for historic reasons? If there was an issue in data generation, shouldn't it also occur in the second set of tests? If there is a distinction and it tests particular integration/diagnostics point ratios, then this would simply need some extra documentation.

Reply:
Yeah I agree with you, randomness in tests can be nasty. The `.seed` has been added.
I have also removed the redundant tests now since I agree they are now better covered by the tests that also check the output from `get_variables_data`.

I also added CHANGELOG.